File: Algorithms/Greedy/Marc-s Cakewalk/MarcsCakewalk.java
import java.util.*;

public class Solution {

    public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		long res = 0;
		int a[] = new int[n];
		for (int i = 0; i < a.length; i++) {
			a[i] = in.nextInt();
		}	
		Arrays.sort(a);
		for (int i = 0; i < a.length; i++) {
			res+= a[a.length-1-i]*Math.pow(2, i);
		}		
		System.out.println(res);
	}
}

File: Algorithms/Greedy/Marc-s Cakewalk/MarcsCakewalk.py
#!/bin/python3

import math
import os
import random
import re
import sys

def marcsCakewalk(calorie):
    res = 0
    calorie = sorted(calorie)
    n = len(calorie)

    for i in range(n):
        res += calorie[n - 1 - i] * math.pow(2, i)

    return int(res)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    calorie = list(map(int, input().rstrip().split()))

    result = marcsCakewalk(calorie)

    fptr.write(str(result) + '\n')

    fptr.close()
